### DISCUSSION TITLE
Feature idea: new helper / utility for returning percentage from two values

### DISCUSSION THREAD

**Discussion by sts-ryan-holton** [laravel/framework#49761] [Ideas]
:wave: Hi,

I'd like to propose a new feature request for being able to return a percentage back when two inputs are given. In my example, I've created a function in my application that's globally accessible but wondered if this kind of function could be ported into a nice [helper or utility](https://laravel.com/docs/10.x/helpers) for Laravel, for example:

Here's my function:

```php
function conversion_from($totals, $conversions, $formatted = true)
{
    if (! $totals || ! $conversions) {
        return 0;
    }

    $conversionRate = round($conversions / $totals, 3) * 100;

    if (! $formatted) {
        return floatval(number_format($conversionRate, 2));
    }

    return number_format(floatval($conversionRate), 2, '.', ',');
}
```

In this example, given the following:

```
var_dump(conversion_from(100, 50)); // returns 50 as int
var_dump(conversion_from(100, 50, true)); // returns 50.00 as string
```

I've created a [demo here](https://onlinephp.io/c/611c2)

I was hoping Laravel could introduce some utility like:

```php
Percentage::from(100, 50) // output 50 as int
Percentage::from(100, 50)->toString() // 50.00 as string
Percentage::from(100, 50)->format('custom format') // handle custom
Percentage::from(100, 50)->toString()->append('%') // 50.00% as string with symbol
Percentage::from(100, 50, 2)->toString() // 50.00 as string with decimal place
```

**Comment by henzeb** [comment]
Sounds more of a Number thing.

**Reply by sts-ryan-holton** [reply]
Sure. So could maybe be a method on the Number utility?

**Comment by shaedrich** [comment]
You'd probably want to
* have an option to pass in the number of decimal places you want to be displayed
* respect locale settings of the decimal separator and the thousands separator

**Reply by sts-ryan-holton** [reply]
Sounds good. Hopefully someone is able to implement

**Reply by ghost** [reply]
@sts-ryan-holton Hi. What value does this bring that compensates for the overhead it comes with? Why is number_format not enough?

**Reply by sts-ryan-holton** [reply]
number_format doesn't do the logical calculations that I intended for this percentage function. Number format simply formats a number. Did you have any methods in Laravel that can do the same division followed by the format?

**Reply by ghost** [reply]
No. That is the point. I used number format for the logic I needed, when I needed it.

**Comment by shaedrich** [comment]
I'm curious about `format()` and `append()`: I mean, are there really so many possibilities that one would allow for passing an arbitrary string?
